GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1551 @@
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError creating new Group {0}", Program.ExtractErrorMessage(e));
+            }
+            return newGroup;
+
+            #endregion
+        }
+
+        private static async Task PrintAllRoles(IActiveDirectoryClient client, string searchString)
+        {
+            #region Get All Roles
+
+            //*********************************************************************
+            // Get All Roles
+            //*********************************************************************
+            List<IDirectoryRole> foundRoles = null;
+            try
+            {
+                IPagedCollection<IDirectoryRole> rolesCollection = await client.DirectoryRoles.ExecuteAsync();
+                foundRoles = rolesCollection.CurrentPage.ToList();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Roles {0}", Program.ExtractErrorMessage(e));
+            }
+
+            if (foundRoles != null && foundRoles.Count > 0)
+            {
+                foreach (IDirectoryRole role in foundRoles)
+                {
+                    Console.WriteLine("\n Found Role: {0} {1} {2} ",
+                        role.DisplayName, role.Description, role.ObjectId);
+                }
+            }
+            else
+            {
+                Console.WriteLine("Role Not Found {0}", searchString);
+            }
+
+            #endregion
+        }
+
+        private static async Task PrintServicePrincipals(IActiveDirectoryClient client)
+        {
+            #region Get Service Principals
+
+            //*********************************************************************
+            // get the Service Principals
+            //*********************************************************************
+            IPagedCollection<IServicePrincipal> servicePrincipals = null;
+            try
+            {
+                servicePrincipals = await client.ServicePrincipals.ExecuteAsync();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Service Principal {0}", e.Message, Program.ExtractErrorMessage(e));
+            }
+            while (servicePrincipals != null)
+            {
+                List<IServicePrincipal> servicePrincipalsList = servicePrincipals.CurrentPage.ToList();
+                foreach (IServicePrincipal servicePrincipal in servicePrincipalsList)
+                {
+                    Console.WriteLine("Service Principal AppId: {0}  Name: {1}", servicePrincipal.AppId,
+                        servicePrincipal.DisplayName);
+                }
+                servicePrincipals = await servicePrincipals.GetNextPageAsync();
+            }
+
+            #endregion
+        }
+
+        private static async Task<string> GetAppObjectId(IActiveDirectoryClient client, string servicePrincipalObjectId)

```

Comment:
>servicePrincipalObjectId [](start = 95, length = 24)

We are looking up ServicePrincipal ObjectId using applicationId. So the input parameter here is applicationId and the name of the method should be GetServicePrincipalObjectId.